projects/kit/directives/dropdown-context/dropdown-context.directive.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    ComponentFactoryResolver,
+    ComponentRef,
+    Directive,
+    ElementRef,
+    forwardRef,
+    HostListener,
+    Inject,
+    Injector,
+    Input,
+    Optional,
+} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {
+    getClosestFocusable,
+    setNativeFocused,
+    TuiActiveZoneDirective,
+    TuiContextWithImplicit,
+    tuiDefaultProp,
+    TuiDestroyService,
+    TuiParentsScrollService,
+    TuiPortalService,
+} from '@taiga-ui/cdk';
+import {AbstractTuiDropdown, TUI_DROPDOWN_DIRECTIVE, TuiDropdown} from '@taiga-ui/core';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+import {fromEvent, merge} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+import {TuiDropdownContextHostComponent} from './dropdown-context-host.component';
+
+// @dynamic
+@Directive({
+    selector: '[tuiDropdownContext]',
+    providers: [
+        TuiParentsScrollService,
+        TuiDestroyService,
+        {
+            provide: TUI_DROPDOWN_DIRECTIVE,
+            useExisting: forwardRef(() => TuiDropdownContextDirective),
+        },
+    ],
+})
+export class TuiDropdownContextDirective<C extends object>
+    extends AbstractTuiDropdown
+    implements TuiDropdown {
+    @Input('tuiDropdownContext')
+    @tuiDefaultProp()
+    content: PolymorpheusContent = '';
+
+    @Input('tuiDropdownContextData')
+    @tuiDefaultProp()
+    context: C | null = null;
+
+    private hostRef: ComponentRef<TuiDropdownContextHostComponent> | null = null;
+
+    private get dropdownContent(): HTMLElement | null {
+        return this.hostRef?.instance.contentRef?.nativeElement || null;
+    }
+
+    constructor(
+        protected readonly elementRef: ElementRef,
+        @Inject(WINDOW) private readonly windowRef: Window,
+        @Inject(TuiDestroyService) readonly destroy$: TuiDestroyService,
+        @Inject(ComponentFactoryResolver)
+        readonly componentFactoryResolver: ComponentFactoryResolver,
+        @Inject(Injector) readonly injector: Injector,
+        @Inject(TuiPortalService) readonly portalService: TuiPortalService,
+        @Inject(TuiParentsScrollService) readonly refresh$: TuiParentsScrollService,
+        @Optional() readonly activeZone: TuiActiveZoneDirective | null,
+    ) {
+        super(componentFactoryResolver, injector, portalService, elementRef, activeZone);
+
+        merge(refresh$, fromEvent(this.windowRef, 'resize'))
+            .pipe(takeUntil(destroy$))
+            .subscribe(() => this.closeDropdown());
+    }
+
+    @HostListener('contextmenu', ['$event'])
```

Comment:
`contextmenu.prevent` instead of explicit call

Reply:
done